### ISSUE TITLE
Curating Resource Event Type.

### ISSUE THREAD

**Issue by vprdnew** [shah/uniform-resource#10]
Hi @shah ,

Event content curation will need the date and location as a mandatory field. We can add this feature by using the schema structure. BY reading schema the date and venue can be taken and added to resource.

**Comment by shah** [comment]
@vprdnew please provide full example code for what you need here so I can add it or you should the code directly and send me a PR.

**Comment by vprdnew** [comment]
HI @shah 

We had executed a query selector to get the schema.

`JSON.parse(document.querySelector(\'script[type="application/ld+json"]\').innerText)`

Below is the code block

```
chrome.tabs.executeScript({
    code: 'JSON.parse(document.querySelector(\'script[type="application/ld+json"]\').innerText)'
}, (selection) => {
    if (selection[0]) {
        let eventInfo = [];
        let contentType = '';
        let flag = 0;
        if (typeof (selection[0][0]) !== 'undefined' && flag === 0) {
            if (typeof (selection[0][0]["@type"]) !== 'undefined') {
                eventInfo = selection[0][0];
                contentType = selection[0][0]["@type"];
                flag = 1;
            }
        } else if (typeof (selection[0]) !== 'undefined' && flag === 0) {
            if (typeof (selection[0]["@type"]) !== 'undefined') {
                eventInfo = selection[0];
                contentType = selection[0]["@type"];
                flag = 1;
            }
        }

        if (flag === 0 && typeof (selection[0]["@graph"]) !== 'undefined') {
            if (typeof (selection[0]["@graph"][0]["@type"]) !== 'undefined') {
                eventInfo = selection[0]["@graph"][0];
                contentType = selection[0]["@graph"][0]["@type"];
                flag = 1;
            }
        }

        if ((contentType.toLowerCase()).includes('event')) {

            if (typeof (eventInfo.startDate) != "undefined") {
                let startDate = new Date(eventInfo.startDate);
                this.setState({ eventStartDate: startDate });
            }

            if (typeof (eventInfo.endDate) != "undefined") {
                let endDate = new Date(eventInfo.endDate);
                this.setState({ eventEndDate: endDate });
            }

            if (typeof (eventInfo.location) != "undefined") {
                let location = eventInfo.location;
                if (location["@type"].toLowerCase() == "place") {
                    this.setState({ EventLocation: location.name });
                }

            }
        }
    }
});
```

The above codes read the schema and find the event startDate, endDate, location and set as state variable.

**Comment by shah** [comment]
This is good input @vprdnew - can you give me the Cheerio selector string too?

**Comment by shah** [comment]
This is implemented in 0.9.1, see test cases for examples.
